Code/Framework/AzToolsFramework/AzToolsFramework/Picking/Manipulators/ManipulatorBounds.cpp

```diff
@@ -16,6 +16,83 @@ namespace AzToolsFramework
 {
     namespace Picking
     {
+        // this intersection algorithm is adapted from 'Capped Cone' by Inigo Quilez
+        // ref: https://www.iquilezles.org/www/articles/intersectors/intersectors.htm and https://www.shadertoy.com/view/llcfRf
+        // all algorithms/code snippets are kindly made available under the MIT License - https://www.iquilezles.org/www/index.htm
+        bool IntersectRayCone(
+            const AZ::Vector3& rayOrigin,
+            const AZ::Vector3& rayDirection,
+            const AZ::Vector3& coneApex,
+            const AZ::Vector3& coneAxis,
+            float coneHeight,
+            float coneBaseRadius,
+            float& t)
+        {
+            const AZ::Vector3& pa = coneApex;
+            const AZ::Vector3& pb = coneApex + coneHeight * coneAxis;
+            const AZ::Vector3& ro = rayOrigin;
+            const AZ::Vector3& rd = rayDirection;
+            float rb = coneBaseRadius;
+
+            AZ::Vector3 ba = pb - pa;
+            AZ::Vector3 oa = ro - pa;
+            AZ::Vector3 ob = ro - pb;
+
+            float m0 = ba.Dot(ba);
+            float m1 = oa.Dot(ba);
+            float m2 = ob.Dot(ba);
+            float m3 = rd.Dot(ba);
+
+            auto dot2 = [](const AZ::Vector3& v)
+            {
+                return v.Dot(v);
+            };
+
+            // caps
+            if (m1 < 0.0f)
+            {
+                if (dot2(oa * m3 - rd * m1) < 0.0f)
+                {
+                    t = -m1 / m3;
+                    return true;
+                }
+            }
+            else if (m2 > 0.0f)
+            {
+                if (dot2(ob * m3 - rd * m2) < (rb * rb * m3 * m3))
+                {
+                    t = -m2 / m3;
+                    return true;
+                }
+            }
+
+            // body
+            float m4 = rd.Dot(oa);
+            float m5 = oa.Dot(oa);
+            float rr = -rb;
+            float hy = m0 + rr * rr;
+
+            float k2 = m0 * m0 - m3 * m3 * hy;
+            float k1 = m0 * m0 * m4 - m1 * m3 * hy;
+            float k0 = m0 * m0 * m5 - m1 * m1 * hy;
+
+            float h = k1 * k1 - k2 * k0;
```

Comment:
h seems like a confusing choice of variable name here since an obvious intuition about what it might mean is height, but it's actually the discriminant in the quadratic equation here.

Reply:
I left the code more or less 'as-is' to be as close to the original as possible, by renaming these variables it will make mapping back to the original more difficult but if we think that trade off is worth it I could update the variable `h` to be `discriminant`